code/modules/power/apc.dm

```diff
@@ -196,12 +197,27 @@
 			pixel_x = -25
 	if (building)
 		area = get_area(src)
+		clear_previous_power_alarm(src, area)
 		opened = APC_COVER_OPENED
 		operating = FALSE
 		name = "\improper [get_area_name(area, TRUE)] APC"
 		set_machine_stat(machine_stat | MAINT)
 		update_appearance()
 		addtimer(CALLBACK(src, .proc/update), 5)
+		area.poweralert(FALSE, src)
+
+/obj/machinery/power/apc/proc/clear_previous_power_alarm(obj/source, area/A)
+	var/list/L = GLOB.alarms["Power"]
+	for (var/I in L)
+		if(I == A.name)
+			var/list/alarm = L[I]
+			var/list/srcs  = alarm[3]
+			for(var/origin in srcs)
+				if(origin != source)//We don't want to clear our own alarm, do we
+					area.poweralert(TRUE, origin)
+					srcs -= origin
+			if (srcs.len == 0)
+				L -= I
```

Comment:
Is this entire proc even necessary? shouldn't `poweralert()` take care of this?

Reply:
Nnnope, poweralert needs an APC as an object, and if the previous APC gets completelly destroyed, it even gets removed from the global APC list sooo there's no way to clear up the power alert tied to it.